test/e2e/gameserver_test.go

```diff
@@ -522,17 +525,39 @@ func TestGameServerWithPortsMappedToMultipleContainers(t *testing.T) {
 	defer framework.AgonesClient.AgonesV1().GameServers(defaultNs).Delete(readyGs.ObjectMeta.Name, nil) // nolint: errcheck
 	assert.Equal(t, readyGs.Status.State, agonesv1.GameServerStateReady)
 
-	firstContainerReply, err := e2eframework.SendGameServerUDPToPort(readyGs, "gameport", "Ping 1")
-	if err != nil {
-		t.Fatalf("Could not message GameServer: %v", err)
-	}
-	assert.Contains(t, firstContainerReply, "Ping 1")
+	interval := 2 * time.Second
+	timeOut := 60 * time.Second
 
-	secondContainerReply, err := e2eframework.SendGameServerUDPToPort(readyGs, "second-gameport", "Ping 2")
-	if err != nil {
-		t.Fatalf("Could not message GameServer: %v", err)
-	}
-	assert.Contains(t, secondContainerReply, "Ping 2")
+	pollCh1 := make(chan error)
+	pollCh2 := make(chan error)
+
+	go func(reply chan error) {
+		errPoll := wait.PollImmediate(3*time.Second, 15*time.Second, func() (done bool, err error) {
+			res, err := e2eframework.SendGameServerUDPToPort(readyGs, firstPort, "Ping 1")
+			if err != nil {
+				t.Logf("Could not message GameServer on %s: %v. Will try again...", firstPort, err)
+			}
+			return err == nil && strings.Contains(res, "Ping 1"), nil
+		})
+		reply <- errPoll
+	}(pollCh1)
+
+	go func(reply chan error) {
+		errPoll := wait.PollImmediate(interval, timeOut, func() (done bool, err error) {
```

Comment:
Strange enough that wait.PollImmediate() parameters are different for similar goroutine functions. Is there an option to declare one function and pass only `firstPort` string and `pollCh1`. Thus we can make code easier to read.

Reply:
I prefer to keep it as is because otherwise, we will have too much function argument